tests/sdk/nodejs/server-side-apply/step2/index.ts

```diff
@@ -0,0 +1,180 @@
+// 1. Patch a Namespace resource with fully-specified configuration.
+// 2. Patch a CustomResource.
+// 3. Upsert a Deployment resource that already exists.
+// 4. Patch the Deployment with a partially-specified configuration.
+// 5. Replace a statically-named ConfigMap resource by changing the data on a subsequent update.
+
+// Create provider with SSA enabled.
+const provider = new k8s.Provider("k8s", {enableServerSideApply: true});
+
+// Create a randomly-named Namespace.
+const ns = new k8s.core.v1.Namespace("test", undefined, {provider});
+
+// Patch the Namespace with additional labels.
+export const nsPatched = new k8s.core.v1.NamespacePatch("test", {
+    metadata: {
+        labels: {
+            foo: "foo",
+        },
+        name: ns.metadata.name,
+    }
+}, {provider});
+
+const crd = new k8s.apiextensions.v1.CustomResourceDefinition("crontab", {
+    metadata: { name: "crontabs.stable.example.com" },
+    spec: {
+        group: "stable.example.com",
+        versions: [
+            {
+                name: "v1",
+                served: true,
+                storage: true,
+                schema: {
+                    openAPIV3Schema: {
+                        type: "object",
+                        properties: {
+                            spec: {
+                                type: "object",
+                                properties: {
+                                    cronSpec: {
+                                        type: "string"
+                                    },
+                                    image: {
+                                        type: "string"
+                                    },
+                                    replicas: {
+                                        type: "integer"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+        ],
+        scope: "Namespaced",
+        names: {
+            plural: "crontabs",
+            singular: "crontab",
+            kind: "CronTab",
+            shortNames: ["ct"]
+        },
+    }
+}, {provider});
+
+// Create a k8s CustomResource of type "CronTab".
+const cr = new k8s.apiextensions.CustomResource("my-new-cron-object",
+    {
+        apiVersion: "stable.example.com/v1",
+        kind: "CronTab",
+        metadata: {
+            name: "my-new-cron-object",
+            namespace: ns.metadata.name,
+        },
+        spec: { cronSpec: "* * * * */6", image: "my-awesome-cron-image" }
+    }, {provider, dependsOn: [crd]});
+
+// Patch the CronTab CustomResource to add a label.
+export const crPatched = new k8s.apiextensions.CustomResourcePatch("label-cr",
+    {
+        apiVersion: "stable.example.com/v1",
+        kind: "CronTab",
+        metadata: {
+            labels: {
+                foo: "foo",
+            },
+            name: "my-new-cron-object",
```

Comment:
Just wanted to make sure we are not letting patch resources by autonamed in the provider? I did notice that we are not enforcing its requirement in the SDK generation (which maybe we ought to do?)

Reply:
Good point. The name should be required.